stdlib/internal/SwiftSyntax/Trivia.swift

```diff
@@ -0,0 +1,279 @@
+      self = .spaces(value)
+    case "Tab":
+      let value = try container.decode(Int.self, forKey: .value)
+      self = .tabs(value)
+    case "VerticalTab":
+      let value = try container.decode(Int.self, forKey: .value)
+      self = .verticalTabs(value)
+    case "Formfeed":
+      let value = try container.decode(Int.self, forKey: .value)
+      self = .formfeeds(value)
+    case "Newline":
+      let value = try container.decode(Int.self, forKey: .value)
+      self = .newlines(value)
+    case "Backtick":
+      let value = try container.decode(Int.self, forKey: .value)
+      self = .backticks(value)
+    case "LineComment":
+      let value = try container.decode(String.self, forKey: .value)
+      self = .lineComment(value)
+    case "BlockComment":
+      let value = try container.decode(String.self, forKey: .value)
+      self = .blockComment(value)
+    case "DocLineComment":
+      let value = try container.decode(String.self, forKey: .value)
+      self = .docLineComment(value)
+    case "DocBlockComment":
+      let value = try container.decode(String.self, forKey: .value)
+      self = .docLineComment(value)
+    default:
+      let context =
+        DecodingError.Context(codingPath: [CodingKeys.kind],
+                              debugDescription: "invalid TriviaPiece kind \(kind)")
+      throw DecodingError.valueNotFound(String.self, context)
+    }
+  }
+  
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    switch self {
+    case .blockComment(let comment):
+      try container.encode("BlockComment", forKey: .kind)
+      try container.encode(comment, forKey: .value)
+    case .docBlockComment(let comment):
+      try container.encode("DocBlockComment", forKey: .kind)
+      try container.encode(comment, forKey: .value)
+    case .docLineComment(let comment):
+      try container.encode("DocLineComment", forKey: .kind)
+      try container.encode(comment, forKey: .value)
+    case .lineComment(let comment):
+      try container.encode("LineComment", forKey: .kind)
+      try container.encode(comment, forKey: .value)
+    case .formfeeds(let count):
+      try container.encode("Formfeed", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .backticks(let count):
+      try container.encode("Backtick", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .newlines(let count):
+      try container.encode("Newline", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .spaces(let count):
+      try container.encode("Space", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .tabs(let count):
+      try container.encode("Tab", forKey: .kind)
+      try container.encode(count, forKey: .value)
+    case .verticalTabs(let count):
+      try container.encode("VerticalTab", forKey: .kind)
+      try container.encode(count, forKey: .value)
```

Comment:
You could use a method that takes the key+count if you think it's worth factoring this.  I don't think gyb'ing is the right answer for this.  Gyb adds a large cognitive burden to any code that uses it, so it has to have a big benefit to be worth it IMO.